### PR TITLE
Fix flaky ping test for remote clusters service

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
On occasion (at least once) the platform/services/remotecluster/ping_test.go units tests would fail.  

This is due to the line https://github.com/mattermost/mattermost/blob/ffc08858cf6344af2855dd5bec4e7a00dd866fd8/server/platform/services/remotecluster/service.go#L264 introduced in PR https://github.com/mattermost/mattermost/pull/26753

All registered remotes are pinged immediately on service startup, whereas before they were pinged via a thread loop asynchronously.  The flaky test is counting the number of pings for one cycle, which now means the initial ping on service start, and on occasion the thread loop kicks in before the test gets shutdown and adds some more pings.

To fix this the unit test has been refactored to achieve the underlying goal of making sure all the remotes get pinged. It doesn't matter if some happen to get multiple pings.

CI fail report:  https://github.com/mattermost/mattermost/runs/23949896037

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-57924

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by wiggin77** [comment]
> With enough attempts I am still able to reproduce some failure.

Ok, my theory is busted.  I'll look deeper.

**Comment by streamer45** [comment]
> BTW, I'm still not able to reproduce, even with 10K iterations. Must be machine specific timings.

Yeah, if it becomes a time sink please let me know and I'll try to spare some cycles to look into it given I can repro every now and then.

**Comment by wiggin77** [comment]
> BTW, I'm still not able to reproduce, even with 10K iterations. Must be machine specific timings.

@streamer45 I figured it out.  I updated the description to match the root cause and the fix.

**Comment by wiggin77** [comment]
I was able to reproduce by running the tests while making my local machine busy with other things.

File: server/platform/services/remotecluster/ping_test.go [hunk 1]
```diff
@@ -77,31 +86,37 @@ func TestPing(t *testing.T) {
 		require.NoError(t, err)
 		defer service.Shutdown()
 
-		wg.Wait()
+		// wait up to 10 seconds for all remotes to get pinged. This will normally take less than 1 second
+		// until the server is very busy.
```
Comment (wiggin77) [inline server/platform/services/remotecluster/ping_test.go:90]:
```suggestion
		// unless the server is very busy.
```

**Review by streamer45 — APPROVED** [review]
Looks good. Thanks for taking care of this!

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+

**Pull request merged**